python/scope.py

```diff
@@ -959,13 +962,14 @@ async def eval_expr(self, expr):
             file_path = os.path.join(os.path.dirname(self.file_path), rel_file_path)
             with open(self.file_path, "r", encoding="utf-8") as f:
                 self.stack_trace.append((expr, File(f, name=os.path.relpath(self.file_path, start=self.base_path))))
+            impn, f = type_check_file(file_path, self.base_path)
```

Comment:
Hmm, this doesn't seem good for performance. It would be quite exponentially scary since type checking the import will type check its imports, and so on, but when you then evaluate the file, that'll type check its imports *again*, which makes things a lot scarier. Especially since you can use `imp` in a loop during runtime.

Perhaps we should have a global dictionary storing a cache of type checking/unit test results from the main type checking stage, and during runtime, it can look up the type checking results during runtime, and perhaps type check the import if there's no entry in the cache.

Reply:
how would one do that though, because unit tests are associated with the `NModule` variable